### ISSUE TITLE
[Programmatic Access - Bot Framework Web Chat - Chat]: Ensures aria-hidden elements are not focusable nor contain focusable elements

### ISSUE THREAD

**Issue by pgopinadhu** [microsoft/BotFramework-WebChat#4720]
### **User Experience:**
Keyboard users and people who rely on assistive technologies and users get affected if the focus lands on the hidden elements while navigating. Users might consider it as an interactive control and try to access them but visually the controls are not visible and hence get confused.
### **Note:**
User credentials should NOT be included in the bug.
### **Repro Steps:**
1. Launch URL: https://microsoft.github.io/BotFramework-WebChat/01.getting-started/a.full-bundle/ on edge dev.
2. Web chat page opens.
3. Press 4-tab keys to reach 'Type your message' field. Enter the "markdown" keyword on it. Press 1-tab key to reach the 'send' button and activate it with the enter key.
4. Press 2-shift tabs and 1-down arrow to reach the received card. Activate it with enter key. Focus lands on 'mark it down' link. Turn on AI- Fast pass.
5. Verify whether the focusable elements are defined as hidden or not.
### **Actual Result:**
The aria-hidden elements are focusable nor contain focusable elements.
### **Expected Result:**
Ensures aria-hidden elements are not focusable nor contain focusable elements (.webchat__keyboard-help).
### **Issue:**
Ensures aria-hidden elements are not focusable nor contain focusable elements (aria-hidden-focus - https://accessibilityinsights.io/info-examples/web/aria-hidden-focus)
### **Target application:**
Web Chat: Full-featured bundle - https://microsoft.github.io/BotFramework-WebChat/01.getting-started/a.full-bundle/
### **Element path:**
.webchat__basic-transcript__terminator
### **Snippet:**
`<div aria-hidden="true" aria-labelledby="webchat__basic-transcript__terminator-label--pk1ws" class="webchat__basic-transcript__terminator" role="note" tabindex="0">`
### **How to fix:**
Fix all of the following:
Focusable content should have tabindex='-1' or be removed from the DOM.
### **Note:**
- Issue reproes on messages/ articles, send section etc on all the articles like (suggested-actions, markdown response, carousel, attach a file etc.)
- Issue reproes throughout the application.

### **MAS** Reference:_
[MAS 4.1.2 – Name, Role, Value](https://aka.ms/MAS4.1.2)
### **Reference Links:**
- [Accessibility Insights!](https://nam06.safelinks.protection.outlook.com/?url=https://accessibilityinsights.io/&data=02%7c01%7cv-poyad%40microsoft.com%7cc11478591b5a474f6d9d08d7199726c0%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c637006009300301864&sdata=ZTWYamT1yE7yko5QU35ZJLf/5ZydtDeGAKrQGUiugJs%3D&reserved=0) - Identify accessibility bugs before check-in and make bug fixing faster and easier.
- External Bug Process: If this bug belongs to external team, mark it as resolved/done and assign it back to the tester with notes on where to file/route the bug. For more information, please use this [Link](https://microsoft.sharepoint.com/teams/cea11y/SitePages/External-Bug-Process.aspx?xsdata=MDV8MDF8fDNkZjhhMWU0NTA0YTQwM2MyNGRhMDhkYWY1NTkwMDNlfDcyZjk4OGJmODZmMTQxYWY5MWFiMmQ3Y2QwMTFkYjQ3fDB8MHw2MzgwOTIwNTk4MzEzMTM2MjZ8VW5rbm93bnxWR1ZoYlhOVFpXTjFjbWwwZVZObGNuWnBZMlY4ZXlKV0lqb2lNQzR3TGpBd01EQWlMQ0pRSWpvaVYybHVNeklpTENKQlRpSTZJazkwYUdWeUlpd2lWMVFpT2pFeGZRPT18MXxNVFkzTXpZd09URTRNall3TXpzeE5qY3pOakE1TVRneU5qQXpPekU1T2pjMllUWXlNRE5pTnpZeFpqUmlObVE0T1dKbE0yRm1aVFZpTjJZd1pHSmhRSFJvY21WaFpDNTJNZz09fDBiZDAxMWU1M2M0YjQ0NGYyNGRhMDhkYWY1NTkwMDNlfGI0NDRlYzhjODQ4YzRiODI4Mzk1NmRjNDdlZjQ3NDky&sdata=dlJVM3lpMWQrRmtpc09HVTI0REh4NlVrU00yMG5xd0JlUFdBOHZuaHNOVT0%3D&ovuser=72f988bf-86f1-41af-91ab-2d7cd011db47%2cv-pgopinadhu%40microsoft.com&OR=Teams-HL&CT=1673616420039&clickparams=eyJBcHBOYW1lIjoiVGVhbXMtRGVza3RvcCIsIkFwcFZlcnNpb24iOiIyNy8yMzAxMDUwOTQwMCIsIkhhc0ZlZGVyYXRlZFVzZXIiOmZhbHNlfQ%3D%3D). “HCL Staff should not log any third-party external bugs. Should be routed to edad team”.
- Please reach out to [C&AI Teams channel](https://nam06.safelinks.protection.outlook.com/?url=https://teams.microsoft.com/l/team/19%253a3cd071821b694f11b054d878453cd75f%2540thread.skype/conversations?groupId%3D30e3f968-15dd-4da2-bd8e-85f27e4af2c3%26tenantId%3D72f988bf-86f1-41af-91ab-2d7cd011db47&data=02%7c01%7cv-dhti%40microsoft.com%7cf90880a914be401871f108d763233116%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c637086876782361405&sdata=/ULS%2BbeqMjysCoiWHvFgAbSnuFJqxO3wtpHmlidPjbU%3D&reserved=0) for any process related queries.
### **Test Environment:**
OS: Windows 11 Enterprise 22H2 (OS build: 22621.1555)
Browser: Edge Version 114.0.1807.6 (Official build) dev (64-bit)
URL: [Web Chat: Full-featured bundle (microsoft.github.io)](https://microsoft.github.io/BotFramework-WebChat/01.getting-started/a.full-bundle/)
Tools: Developer Tool, Accessibility Insights for Web
Matrix: Edge dev+ Narrator.

**Comment by pgopinadhu** [comment]
#A11yMAS;#A11ySev2;#MAS4.1.2;#AIConfirmed;HCL;#Accessibility;#Benchmark;#BotFrameworkWebChat-Apr23;
#Win11-Edge(Chromium)

**Comment by compulim** [comment]
Fixed in #4652 and released in 4.15.8.

**Comment by pgopinadhu** [comment]
As verified the issue on https://microsoft.github.io/BotFramework-WebChat/01.getting-started/a.full-bundle/ under Windows 11 Enterprise 22H2 (OS build: 22621.1702), Edge Version 115.0.1851.0 (Official build) dev (64-bit) and observed that the issue is fixed. Hence, Closing the bug.
